griptape/artifacts/text_artifact.py

```diff
@@ -11,23 +11,36 @@
     from griptape.tokenizers import BaseTokenizer
 
 
+def value_to_str(value: Any) -> str:
+    if isinstance(value, dict):
+        return "\n".join(f"{key}: {val}" for key, val in value.items())
+    else:
+        return str(value)
+
+
 @define
 class TextArtifact(BaseArtifact):
-    value: str = field(converter=str, metadata={"serializable": True})
+    value: str = field(converter=value_to_str, metadata={"serializable": True})
     encoding: str = field(default="utf-8", kw_only=True)
     encoding_error_handler: str = field(default="strict", kw_only=True)
     _embedding: list[float] = field(factory=list, kw_only=True)
 
-    @property
-    def embedding(self) -> Optional[list[float]]:
-        return None if len(self._embedding) == 0 else self._embedding
-
     def __add__(self, other: BaseArtifact) -> TextArtifact:
         return TextArtifact(self.value + other.value)
 
     def __bool__(self) -> bool:
         return bool(self.value.strip())
 
+    @property
+    def embedding(self) -> Optional[list[float]]:
+        return None if len(self._embedding) == 0 else self._embedding
```

Comment:
should we just default `_embedding` to `None` and rename the field?

Reply:
Yeah I'm not really sure what this pattern gives us, besides having to check for `None`.

Reply:
What does that get us specifically? I think the intended behavior here is to distinguish between non-generated vs empty embeddings, right?

Reply:
its just less verbose and would allow us to remove the `@property` method while accomplishing the same thing

Reply:
```
embedding: Optional[list[float]] = field(default=None, kw_only=True)
```
should be functionally equivalent to this `@property` no?

Reply:
Updated